src/sampling.jl

```diff
@@ -276,6 +287,18 @@ end
 
 sample(a::AbstractArray) = a[randi(length(a))]
 
+
+"""
+    sample!(a, x, [replace=true,] [ordered=false])
+
+Draw a random sample of elements from an array `a` and store the result
+in `x`. A polyalgorithm is used for sampling. The function accepts two
+keyword arguments:
+
+  • `replace`: Should sampling be performed with replacement? Defaults
+    to `true`.
+  • `ordered`: Should an ordered sample be taken? Defaults to `false`.
```

Comment:
What does "ordered" imply? Just sorting the result (but more efficient)?

Also, as above, go straight to the point.